docs/docs/how_to/debugger/debugging_with_vs_code.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Using the VS Code Debugger
+description:
+  Step by step guide on how to debug your Noir circuits with the VS Code Debugger configuration and features.
+keywords:
+  [
+    Nargo,
+    Noir CLI,
+    Noir Debugger,
+    VS Code,
+    IDE,
+  ]
+sidebar_position: 0
+---
+
+This guide will show you how to use VS Code with the vscode-noir extension to debug a Noir project. 
+
+#### Pre-requisites
+
+- Noir > x
+- Nargo > x
+- vscode-noir > x
+- A Noir project with a `Nargo.toml`, `Prover.toml` and at least one Noir (`.nr`) containing an entry point function (typically `main`).
+
+Once you have installed Nargo and vscode-noir, you should have everything you need to debug your project. 
+
+## Running the debugger
+
+The easiest way to start debugging is to open the file you want to debug, and press `F5`. This will cause the debugger to launch, using your `Prover.toml` file as input.
+
+You should see something like this:
+
+![Debugger launched](@site/static/img/debugger/1-started.png)
+
+Let's inspect the state of the program. For that, we open VS Code's _Debug pane_. Look for this icon:
+
+![Debug pane icon](@site/static/img/debugger/2-icon.png)
+
+You will now see three categories of variables: Locals, Witness Map and Brillig Registers. 
+
+![Debug pane expanded](@site/static/img/debugger/3-debug-pane.png)
+
+1. **Locals**: variables of your program. At this point in execution this section is empty, but as we step through the code it will get populated by `x`, `result`, `digest`, etc. 
+
+2. **Witness map**: these are initially populated from your project's `Prover.toml` file. In this example, they will be used to populate `x` and `result` at the beginning of the `main` function.
+
+3. **Brillig registers**: these show the current state of the BrilligVM registers. 
```

Comment:
Same for the Brilling naming, just keeping in mind (I won't flag this every time I see it moving further on this review)

Reply:
Same comment as above, I see how to use `unconstrained` instead Brillig when talking concepts, but for VM components such as registers it becomes a bit trickier. 

What would you call "Brillig registers"? Unconstrained VM registers? Unconstrained runtime registers?

Reply:
Unconstrained VM registers looks ok... @Savio-Sou any thoughts?